tests/unit/steps/test_yaml_parser.py

```diff
@@ -9,13 +9,15 @@
 from buildbot.process.results import FAILURE, SUCCESS
 from buildbot.test.fake.remotecommand import Expect, ExpectRemoteRef
 from buildbot.test.util import steps
+from buildbot.test.util.misc import TestReactorMixin
 from twisted.trial import unittest
 
 from eve.steps import yaml_parser
 from tests.util.yaml_factory import RawYaml, SingleCommandYaml
 
 
-class TestReadConfFromYamlExecution(steps.BuildStepMixin, unittest.TestCase):
+class TestReadConfFromYamlExecution(steps.BuildStepMixin, TestReactorMixin,
```

Comment:
`TestReactorMixin` 
What is the purpose ​of this class ?  I have seen a lot of code using thins but, i don´t see what it does.

Reply:
You have to use TestReactorMixin in recent buildbot versions otherwise you do no have a reactor function for your testcase object. Basically, previous code just did not work anymore with the version of builbot. Here's why I changed this. And the builtin tests case within buildbot work this way